src/Entity/PolicyListBuilder.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Drupal\password_enhancements\Entity;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityListBuilder;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Url;
+use Drupal\password_enhancements\Entity\Storage\PolicyEntityStorageInterface;
+use Drupal\user\RoleStorageInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * List builder for password policy config entities.
+ */
+class PolicyListBuilder extends EntityListBuilder {
+
+  /**
+   * Role storage.
+   *
+   * @var \Drupal\user\RoleStorageInterface
+   */
+  protected $roleStorage;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(EntityTypeInterface $entity_type, PolicyEntityStorageInterface $storage, RoleStorageInterface $role_storage) {
+    parent::__construct($entity_type, $storage);
+
+    $this->roleStorage = $role_storage;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function createInstance(ContainerInterface $container, EntityTypeInterface $entity_type): PolicyListBuilder {
+    return new static(
+      $entity_type,
+      $container->get('entity.manager')->getStorage($entity_type->id()),
+      $container->get('entity_type.manager')->getStorage('user_role')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function load(): array {
+    $entity_ids = $this->storage->getQuery()->sort('priority', 'asc')->execute();
+    return $this->storage->loadMultiple($entity_ids);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function render(): array {
+    $build = parent::render();
+    $build['table']['#empty'] = $this->t('There is no policy created yet.');
+    return $build;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildHeader(): array {
+    $header = [
+      'policy' => $this->t('Policy'),
+      'minimum_required_constraints' => $this->t('Minimum required constraints'),
+      'expire_days' => $this->t('Password expiry'),
+      'priority' => $this->t('Priority'),
+    ];
+
+    return $header + parent::buildHeader();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDefaultOperations(EntityInterface $entity) {
+    $operations = parent::getDefaultOperations($entity);
+    $operations['manage_constraints'] = [
+      'type' => 'link',
+      'title' => $this->t('Manage constraints'),
+      'url' => Url::fromRoute('entity.password_enhancements_constraint.collection', [
```

Comment:
Since you have added : 

```php
<?php
  /**
   * {@inheritdoc}
   */
  public function urlRouteParameters($rel): array {
    $uri_route_parameters = parent::urlRouteParameters($rel);

    if (!empty($this->policy)) {
      $uri_route_parameters['password_enhancements_policy'] = $this->policy;
    }

    return $uri_route_parameters;
  }
```
to the Constraint entity you should be able to use `$this->entity->toUrl('collection')` here.

Reply:
It does not.

Reply:
Okay so it some places it works and on some it doesn't I didn't wanted to dive deep into why, but it doesn't hurt getting the route like this here.

Reply:
#8